sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -2269,10 +2270,13 @@ public <T> Mono<FeedResponse<T>> readMany(
                                 List<T> finalList = new ArrayList<>();
                                 HashMap<String, String> headers = new HashMap<>();
                                 ConcurrentMap<String, QueryMetrics> aggregatedQueryMetrics = new ConcurrentHashMap<>();
+                                List<ClientSideRequestStatistics> aggregateRequestStatistics = new ArrayList<>();
+                                QueryInfo.QueryPlanDiagnosticsContext firstQueryPlanDiagnosticContext = null;
                                 double requestCharge = 0;
                                 for (FeedResponse<Document> page : feedList) {
                                     ConcurrentMap<String, QueryMetrics> pageQueryMetrics =
                                         ModelBridgeInternal.queryMetrics(page);
+                                    FeedResponseDiagnostics pageDiagnostics = page.getCosmosDiagnostics().getFeedResponseDiagnostics();
                                     if (pageQueryMetrics != null) {
                                         pageQueryMetrics.forEach(
                                             aggregatedQueryMetrics::putIfAbsent);
```

Comment:
Is it a concern that means that you will only get one per key, so metrics from later responses may get dropped?  Do we need this to become a multi-map somehow?

Reply:
looks like this is outdated and no longer relevant, resolving this conversation. In case it is still relevant, we can discuss it.